optuna/structs.py

```diff
@@ -264,58 +264,3 @@ class StudySummary(
         datetime_start:
             Datetime where the :class:`~optuna.study.Study` started.
     """
-
-
-class OptunaError(Exception):
-    """Base class for Optuna specific errors."""
-
-    pass
-
-
-class TrialPruned(OptunaError):
```

Comment:
This changes will break most of the code which uses pruners. So I recommend you to keep backward compatibility with deprecation warnings for a while.

Reply:
That makes sense. It's important to keep user programs executable.
I'll revert the exceptions in `optuna.exceptions` with deprecation warnings.
Thank you.

Reply:
I reverted `TrialPruned` in the commit https://github.com/pfnet/optuna/pull/691/commits/6a12b673a8588c26d923fcd2143bbb51947e36e0. But I changed my mind and stopped to revert the others.

This is because
1) they are not raised in user programs, 
2) I guess quite a few users handle `CLIUsageError`, `StorageInternalError` or `DuplicatedStudyError` in the user programs, and
3) such users are able to migrate the code.

If you have any ideas to preserve all exceptions in `optuna.structs`, please let me know.

Reply:
I agree your opinion. I think it's enough to just keep the compatibility of `TrialPruned` 👍

Reply:
Thank you for your prompt reply 🙇